### ISSUE TITLE
IE11 - ERROR TypeError: Object doesn't support property or method 'entries'

### ISSUE THREAD

**Issue by anuragaroraaa** [ngneat/dialog#12]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior

In Angular v8, 9, 11 ngneat/dialog is throwing error on IE11 
`ERROR TypeError: Object doesn't support property or method 'entries' #12`

when using `const ref = this.dialog.open(TestDialogComponent);`
I tried adding these in polyfills
```
import 'core-js/es';
import 'core-js/features/reflect';
```

but then I am getting 
`ERROR TypeError: Unable to get property 'apply' of undefined or null reference
`

## Expected behavior
This statement `const ref = this.dialog.open(TestDialogComponent);` should work on IE 11 and open dialog successfully with Angular versions 8 or more

## Minimal reproduction of the problem with instructions

1. Create a new Angular project (v8, v9 or latest)
2. Write function to Open dialog with TestDialogComponent( I've taken code reference from @ngneat/dialog example application)
```
openDialog() {

    const ref = this.dialog.open(TestDialogComponent);

    ref.backdropClick$.subscribe({
      next: () => (this.backDropClicked = true)
    });

    return ref;
 }
```
 3. Set "target": "es5" in tsconfig.json
 4. remove not from browserslist from IE11
 5. On firing any event to call the function on Step 2, I am getting this error `ERROR TypeError: Object doesn't support property or method 'entries' ` and dialog is not getting opened
It is not working with this specific use case when dialog gets opened with Component

## Environment

<pre><code>
Angular version: 11.0.5

Browser:
- [ ] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [x] IE version 11
- [ ] Edge version XX
 

</code></pre>

**Comment by anuragaroraaa** [comment]
The suggested solution would be to 
https://github.com/ngneat/dialog/blob/085c361ce2f4bb8a1c70cfc2a81a041a41f9cc8e/projects/ngneat/dialog/src/lib/dialog.component.ts#L87
make this statement work for IE11

**Comment by tonivj5** [comment]
Hi @anuragaroraaa! The spread operator is downgraded by typescript to work on old browsers (so it isn't a problem).

And you should polyfill `object`'s `entries` method in `polyfills.ts` file:
```ts
import 'core-js/features/object/entries';
```

Once done, dialog should work 👍🏻

**Comment by anuragaroraaa** [comment]
Hi @tonivj5, as stated above, I tried adding polyfills for object's entries and was able to fix below error

`ERROR TypeError: Object doesn't support property or method 'entries'`

but as soon as I am adding entries polyfills
I am getting this error

`ERROR TypeError: Unable to get property 'apply' of undefined or null reference`

I tried all the combinations of polyfills for fixing this issue, but nothing worked

Finally it is worked on IE only when I replaced
`host.append(...this.nodes);`

with

`host.appendChild.apply(host, this.nodes);`

Please do suggest some alternatives.

**Comment by Lonli-Lokli** [comment]
@anuragaroraaa add an Object.assign polyfill

**Comment by anuragaroraaa** [comment]
@Lonli-Lokli  I tried that too, same error

`ERROR TypeError: Unable to get property 'apply' of undefined or null reference`

**Comment by tonivj5** [comment]
The problem isn't spread operator itself, the problem is `append` method, [it isn't supported in IE](https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/append#Browser_compatibility)

> Finally it worked on IE only when I replaced
> host.append(...this.nodes);
> with
> host.appendChild.apply(host, this.nodes);

It works because you are using `appendChild` instead of `append`, but it doesn't do the same. If `nodes` has more than one element, it didn't append it (it happens with ng-templates).

As a workaround [you can polyfill `append`](https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/append#Polyfill).

Anyway I have open a PR that should fix it #13 👍🏻 

By the way, `entries` polyfill should be being added if it isn't supported by IE11

**Comment by anuragaroraaa** [comment]
Thanks @tonivj5. Looking forward for the release.
